cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/ShowFunctionsTest.scala

```diff
@@ -28,63 +28,158 @@ class ShowFunctionsTest extends DocumentingTest {
+====
+
+.List functions output
+[options="header", cols="4,6"]
+||===
+|| Column
+|| Description
+
+|m| name
+|a| The name of the function. label:default-output[]
+
+|m| category
+|a| The function category, for example scalar or string. label:default-output[]
+
+|m| description
+|a| The function description. label:default-output[]
+
+|m| signature
+|a| The signature of the function.
+
+|m| isBuiltIn
+|a| Whether the function is built-in or user-defined.
+
+|m| argumentDescription
+|a| List of the arguments for the function, as map of strings with name, type, default, and description.
+
+|m| returnDescription
+|a| The return value type.
+
+|m| aggregating
+|a| Whether the function is aggregating or not.
+
+|m| rolesExecution
+|a|
+List of roles permitted to execute this function.
+Will be `null` without the <<administration-security-administration-dbms-privileges-role-management,`SHOW ROLE`>> privilege.
+
+|m| rolesBoostedExecution
+|a|
+List of roles permitted to use boosted mode when executing this function.
+Will be `null` without the <<administration-security-administration-dbms-privileges-role-management,`SHOW ROLE`>> privilege.
+||===""")
+    section("Syntax") {
+      p("""
+List functions, either all or only built-in or user-defined::
+
+[source, cypher, role=noplay]
+----
+SHOW [ALL|BUILT IN|USER DEFINED] FUNCTION[S]
+[YIELD { * | field[, ...] } [ORDER BY field[, ...]] [SKIP n] [LIMIT n]]
+[WHERE expression]
+[RETURN field[, ...] [ORDER BY field[, ...]] [SKIP n] [LIMIT n]]
+----
+
+[NOTE]
+====
+When using the `RETURN` clause, the `YIELD` clause is mandatory and may not be omitted.
+====
+
+List functions that the current user can execute::
+
+[source, cypher, role=noplay]
+----
+SHOW [ALL|BUILT IN|USER DEFINED] FUNCTION[S] EXECUTABLE [BY CURRENT USER]
+[YIELD { * | field[, ...] } [ORDER BY field[, ...]] [SKIP n] [LIMIT n]]
+[WHERE expression]
+[RETURN field[, ...] [ORDER BY field[, ...]] [SKIP n] [LIMIT n]]
+----
+
+[NOTE]
+====
+When using the `RETURN` clause, the `YIELD` clause is mandatory and may not be omitted.
+====
+
+List functions that the specified user can execute::
+
+[source, cypher, role=noplay]
+----
+SHOW [ALL|BUILT IN|USER DEFINED] FUNCTION[S] EXECUTABLE BY username
+[YIELD { * | field[, ...] } [ORDER BY field[, ...]] [SKIP n] [LIMIT n]]
+[WHERE expression]
+[RETURN field[, ...] [ORDER BY field[, ...]] [SKIP n] [LIMIT n]]
+----
+
+Required privilege <<administration-security-administration-dbms-privileges-user-management,`SHOW USER`>>.
+This command cannot be used for LDAP users.
+
+[NOTE]
+====
+When using the `RETURN` clause, the `YIELD` clause is mandatory and may not be omitted.
+====""".stripMargin('#'))
+    }
+    section("Listing all functions") {
+      p("""To list all available functions with the default output columns, the `SHOW FUNCTIONS` command can be used.
+          #If all columns are required, use `SHOW FUNCTIONS YIELD *`.""".stripMargin('#'))
+      query("SHOW FUNCTIONS", ResultAssertions(p => {
```

Comment:
All descriptions render as code blocks and within quotation marks. It's not strictly wrong, but it doesn't look so nice.